pkg/provider/provider.go

```diff
@@ -574,7 +574,8 @@ func (k *kubeProvider) Diff(
 	var isInputPatch bool
 	var patchBase *unstructured.Unstructured
 
-	tryDryRun := supportsDryRun && oldInputs.GroupVersionKind().String() == gvk.String()
+	tryDryRun := supportsDryRun && oldInputs.GroupVersionKind().String() == gvk.String() &&
```

Comment:
We should add a TODO and an issue here tracking that this is not the permanent solution.